### ISSUE TITLE
Review dependencies and remove the ones without concrete tag

### ISSUE THREAD

**Issue by masih** [filecoin-project/lotus#12503]
There are quite a few "wild" dependencies in Lotus [go module](https://github.com/filecoin-project/lotus/blob/master/go.mod); ones that:
* have no concrete tag.
* have unclear cost/benefit.

- [ ] Review the dependencies without concrete tag.
- [ ] Remove them if possible in cases where the usage is minimal and better alternatives exist.
- [ ] Cut a tag or upgrade anything else that's hard to get rid of.

This issue is done when have no wild dependency in Lotus go mod, or there should be a comment above the ones we can't get rid of explaining why.

**Comment by BigLep** [comment]
@masih : I agree this would be done.  To ensure we don't backslide or make the problem worse, I think we need to do https://github.com/filecoin-project/lotus/issues/7131

**Comment by rjan90** [comment]
Posting here since it came up in a [review/comment](https://github.com/filecoin-project/lotus/pull/12524/files/b5fb8b521b006bd8700131e11932b00dc1708a66#r1780528101): Check if we can get rid of the filecoin-ffi dependency in the `require` section in the go.mod file:

https://github.com/filecoin-project/lotus/blob/d017c15cb34edfe7757c8089afbc915f8255ce76/go.mod#L40-L41

Since we already ask it to use the local directory https://github.com/filecoin-project/lotus/blob/d017c15cb34edfe7757c8089afbc915f8255ce76/go.mod#L13-L14

**Comment by rjan90** [comment]
Idea is to get all dependencies to be on a tag, and clean up the dependency tree.
